File: Initial Prototype/ServerSide/FindNDrive/DomainObjects/Domains/PrivacySettings.cs
// --------------------------------------------------------------------------------------------------------------------
// <copyright file="PrivacySettings.cs" company="">
//   
// </copyright>
// <summary>
//   Defines the PrivacySettings type.
// </summary>
// --------------------------------------------------------------------------------------------------------------------

namespace DomainObjects.Domains
{
    using System.ComponentModel.DataAnnotations;
    using System.Runtime.Serialization;
    using DomainObjects.Constants;

    /// <summary>
    /// The privacy settings.
    /// </summary>
    [DataContract]
    public class PrivacySettings
    {
        /// <summary>
        /// Gets or sets the privacy settings id.
        /// </summary>
        [ConcurrencyCheck]
        public int PrivacySettingsId { get; set; }

        /// <summary>
        /// Gets or sets the email privacy level.
        /// </summary>
        [DataMember]
        public PrivacyLevel EmailPrivacyLevel { get; set; }

        /// <summary>
        /// Gets or sets the gender privacy level.
        /// </summary>
        [DataMember]
        public PrivacyLevel GenderPrivacyLevel { get; set; }

        /// <summary>
        /// Gets or sets the date of birth privacy level.
        /// </summary>
        [DataMember]
        public PrivacyLevel DateOfBirthPrivacyLevel { get; set; }

        /// <summary>
        /// Gets or sets the phone number privacy level.
        /// </summary>
        [DataMember]
        public PrivacyLevel PhoneNumberPrivacyLevel { get; set; }

        /// <summary>
        /// Gets or sets the rating privacy level.
        /// </summary>
        [DataMember]
        public PrivacyLevel RatingPrivacyLevel { get; set; }

        /// <summary>
        /// Gets or sets the journeys privacy level.
        /// </summary>
        [DataMember]
        public PrivacyLevel JourneysPrivacyLevel { get; set; }
    }
}

File: Project/Client/MyApplicationProject/MyApplication/src/main/java/com/example/myapplication/domain_objects/PrivacySettings.java
package com.example.myapplication.domain_objects;

/**
 * Created by Michal on 06/03/14.
 */
public class PrivacySettings
{
    private int EmailPrivacyLevel;
    private int GenderPrivacyLevel;
    private int DateOfBirthPrivacyLevel;
    private int PhoneNumberPrivacyLevel;
    private int RatingPrivacyLevel;
    private int JourneysPrivacyLevel;

    public int getJourneysPrivacyLevel() {
        return JourneysPrivacyLevel;
    }

    public int getRatingPrivacyLevel() {
        return RatingPrivacyLevel;
    }

    public int getPhoneNumberPrivacyLevel() {
        return PhoneNumberPrivacyLevel;
    }

    public int getDateOfBirthPrivacyLevel() {
        return DateOfBirthPrivacyLevel;
    }

    public int getGenderPrivacyLevel() {
        return GenderPrivacyLevel;
    }

    public int getEmailPrivacyLevel() {
        return EmailPrivacyLevel;
    }

    public PrivacySettings(int emailPrivacyLevel, int genderPrivacyLevel, int dateOfBirthPrivacyLevel, int phoneNumberPrivacyLevel, int ratingPrivacyLevel, int journeysPrivacyLevel) {
        EmailPrivacyLevel = emailPrivacyLevel;
        GenderPrivacyLevel = genderPrivacyLevel;
        DateOfBirthPrivacyLevel = dateOfBirthPrivacyLevel;
        PhoneNumberPrivacyLevel = phoneNumberPrivacyLevel;
        RatingPrivacyLevel = ratingPrivacyLevel;
        JourneysPrivacyLevel = journeysPrivacyLevel;
    }
}
